src/Symfony/Component/Messenger/Envelope.php

```diff
@@ -48,22 +48,26 @@ public function with(StampInterface ...$stamps): self
         $cloned = clone $this;
 
         foreach ($stamps as $stamp) {
-            $cloned->stamps[\get_class($stamp)] = $stamp;
+            $cloned->stamps[\get_class($stamp)][] = $stamp;
         }
 
         return $cloned;
     }
 
     public function get(string $stampFqcn): ?StampInterface
```

Comment:
As this PR is a BC break, why not always returning an array of all stamps here?

Reply:
I think the use case here is very similar to HeaderBag::get() : you get only one header with, and that's nice for DX as it saves you from the extra code to deal with a collection for headers/stamps that are single.
Then, if you know a specific header/stamp is a collection, then you can get it with a different signature ->all($stampFqcn)
I think that's nice for DX so I'd suggest to keep as is.
Makes sense?

Reply:
`Envelope::get()` has been renamed `Envelope::getLast()`

Reply:
Considering how the other methods of this class are named, shouldn't `Envelope::last()` be more appropriate ?

Reply:
Why not, no specific opinion on this one. I'll update if more people think last() would be better than getLast().

Reply:
Indeed, `last` sounds better.

Reply:
last() it is, updated. Thanks.